Artifacts/windows-chocolatey/ChocolateyPackageInstaller.ps1

```diff
@@ -0,0 +1,262 @@
+
+
+    Known issues / Caveats
+    ======================
+    
+    - No known issues.
+
+
+    Coming soon / planned work
+    ==========================
+
+    - N/A.    
+
+##################################################################################################>
+
+#
+# Optional arguments to this script file.
+#
+
+Param(
+    # comma or semicolon separated list of chocolatey packages.
+    [ValidateNotNullOrEmpty()]
+    [string]
+    $RawPackagesList
+)
+
+##################################################################################################
+
+#
+# Powershell Configurations
+#
+
+# Note: Because the $ErrorActionPreference is "Stop", this script will stop on first failure.  
+$ErrorActionPreference = "stop"
+
+###################################################################################################
+
+#
+# Custom Configurations
+#
+
+$ChocolateyPackageInstallerFolder = Join-Path $env:ALLUSERSPROFILE -ChildPath $("ChocolateyPackageInstaller-" + [System.DateTime]::Now.ToString("yyyy-MM-dd-HH-mm-ss"))
+
+# Location of the log files
+$ScriptLog = Join-Path -Path $ChocolateyPackageInstallerFolder -ChildPath "ChocolateyPackageInstaller.log"
+$ChocolateyInstallLog = Join-Path -Path $ChocolateyPackageInstallerFolder -ChildPath "ChocolateyInstall.log"
+
+##################################################################################################
+
+# 
+# Description:
+#  - Displays the script argument values (default or user-supplied).
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - Please ensure that the Initialize() method has been called at least once before this 
+#    method. Else this method can only write to console and not to log files. 
+#
+
+function DisplayArgValues
+{
+    WriteLog "========== Configuration =========="
+    WriteLog $("RawPackagesList : " + $RawPackagesList)
+    WriteLog "========== Configuration =========="
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Creates the folder structure which'll be used for dumping logs generated by this script and
+#    the logon task.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function InitializeFolders
+{
+    if ($false -eq (Test-Path -Path $ChocolateyPackageInstallerFolder))
+    {
+        New-Item -Path $ChocolateyPackageInstallerFolder -ItemType directory | Out-Null
+    }
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Writes specified string to the console as well as to the script log (indicated by $ScriptLog).
+#
+# Parameters:
+#  - $message: The string to write.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function WriteLog
+{
+    Param(
+        <# Can be null or empty #> $message
+    )
+
+    $timestampedMessage = $("[" + [System.DateTime]::Now + "] " + $message) | % {  
```

Comment:
I don't think so in this case because we are logging locally to the guest machine and it will make sense for the log times to match the VM time.